doc/sphinx/source/users-guide/taskpar/cobegin.rst

```diff
@@ -0,0 +1,87 @@
+:title: Users Guide: The **cobegin** statement
+
+The **cobegin** statement
+=========================
+
+The **cobegin statement** creates a fixed number of related, yet
+potentially heterogeneous, tasks.  It is expressed by prefixing a
+compound statement with the ``cobegin`` keyword, as illustrated by the
+following example:
+
+.. literalinclude:: examples/users-guide/taskpar/cobegin.chpl
+  :caption:
+  :language: chapel
+
+At execution time, a distinct task is created for each child statement
+within the cobegin statement.  Thus, in the example above, three tasks
+will be created, corresponding to the three statements immediately
+within the cobegin's block statement: one for the standalone
+writeln(), one for the call to ``sayHi()``, and one for the compound
+statement.  Since these tasks are likely to execute in parallel and
+don't coordinate with one another, the ordering of their messages
+relative to one another is arbitrary.  Note that the two statements
+within the compound statement will always be printed in the order
+shown since they are executed by a single task.
+
+Cobegin statements also specify that the original task which
+encounters it cannot proceed until its child tasks have completed.  In
+the example above, this means that "See you later!"  will not be
+printed until all of the greetings have been.
+
+As a result of these rules, possible outputs for the code above
+include:
+
+.. literalinclude:: examples/users-guide/taskpar/cobegin.good
+  :language: text
+
+or
+
+.. literalinclude:: examples/users-guide/taskpar/cobegin-alt.good
+  :language: text
+
+but never:
+
+.. literalinclude:: examples/users-guide/taskpar/cobegin-bad.txt
+  :language: text
+
+(because this would imply that the third task did not run serially and
+that the original task did not wait for its children to complete
+before going on).
```

Comment:
You could also add a bad case where "See you later!" is printed before any of the greetings.